### PR TITLE
vc1parser: Fix quantizer setting issue

### PR DESCRIPTION
**Pull request by YuJiankang**
SA00053.vc1 and SA10095.vc1 should be fixed

Signed-off-by: jkyu jiankang.yu@intel.com

### CODE REVIEW

File: codecparsers/vc1Parser.cpp [hunk 1]
```diff
@@ -711,6 +711,8 @@ namespace VC1 {
             m_frameHdr.pquant = m_frameHdr.pqindex;
             if (m_seqHdr.quantizer == 1)
                 m_frameHdr.pquantizer = br->read(1);
+            else if (m_seqHdr.quantizer == 2)
```
Comment (xuguangxin) [inline codecparsers/vc1Parser.cpp:713]: Do we need handle other quantizer? or we need add  some check about this, like:

```
if (m_seqHdr.quantizer > 2)
    return false
```

**Comment by YuJiankang** [comment]
No need return false, for other quantizer, it will use the initial value which equals to 1.

**Comment by xuguangxin** [comment]
We read two bits from bit stream. It should have 4 values, 0, 1, 2, 3. We only handle 0, 1, 2 here. What about 3?

**Comment by YuJiankang** [comment]
m_frameHdr.pquantizer = 1;
        if (m_seqHdr.quantizer == 0) {
            m_frameHdr.pquant = QuantizerTranslationTable[m_frameHdr.pqindex];
            m_frameHdr.pquantizer = (m_frameHdr.pqindex <= 8) ? 1 : 0;
        }
        else {
            m_frameHdr.pquant = m_frameHdr.pqindex;
            if (m_seqHdr.quantizer == 1)
                m_frameHdr.pquantizer = br->read(1);
            else if (m_seqHdr.quantizer == 2)
                m_frameHdr.pquantizer = 0;
        }
the m_frameHdr.pquantizer has initial value 1. if m_seqHdr.quantizer == 3, No change to m_frameHdr.pquantizer, uses the initial value 1.

**Comment by xuguangxin** [comment]
So default value  1 is for m_seqHdr.quantizer == 3 only. Since yo handled every possible value of m_seqHdr.quantizer, Do you think it's ok to change like this.

if (m_seqHdr.quantizer == 0) {
m_frameHdr.pquant = QuantizerTranslationTable[m_frameHdr.pqindex];
m_frameHdr.pquantizer = (m_frameHdr.pqindex <= 8) ? 1 : 0;
}
else {
m_frameHdr.pquant = m_frameHdr.pqindex;
if (m_seqHdr.quantizer == 1)
m_frameHdr.pquantizer = br->read(1);
else if (m_seqHdr.quantizer == 2)
m_frameHdr.pquantizer = 0;
else if (m_seqHdr.quantizer == 3)
m_frameHdr.pquantizer = 1;
else
assert(0 && "bug: invalid quantizer");
}

**Comment by xuguangxin** [comment]
Hi @YuJiankang 
Any comments on this?

**Comment by YuJiankang** [comment]
patch updated, please review again, if no problem help +1, thanks a lot

**Comment by xuguangxin** [comment]
+1

**Pull request merged**